packages/core/core-flows/src/cart/workflows/complete-cart.ts

```diff
@@ -111,22 +111,7 @@ export const completeCartWorkflow = createWorkflow(
       name: "cart-query",
     })
 
-    // this is only run when the cart is completed for the first time (same condition as below)
-    // but needs to be before the validation step
-    const paymentSessions = when(
-      "create-order-payment-compensation",
-      { orderId },
-      ({ orderId }) => !orderId
-    ).then(() => {
-      const paymentSessions = validateCartPaymentsStep({ cart })
-      // purpose of this step is to run compensation if cart completion fails
-      // and tries to cancel or refund the payment depending on the status.
-      compensatePaymentIfNeededStep({
```

Comment:
comment: @fPolic, @carlos-r-l-rodrigues I believe this is not needed if the payment authorization is the final step. Please correct me if I am wrong.

Reply:
It depends on how we want to handle errors with `autocapture` flow.
If the cart complete flow is triggered from the webhook on payment success, and something fails before the authorize step, there would be a payment but order wouldn't be created right?

Reply:
The issue here is what Frane said, when the Payment was authorized/captured before and then this workflow is triggered by the webhook. We have to remember that the "authorize step" in practice is only reading a status from the provider, and not doing any action per se.
At that point, anything that goes wrong in this workflow, we should compensate the payment.

Reply:
> If the cart complete flow is triggered from the webhook on payment success, and something fails before the authorize step, there would be a payment but order wouldn't be created right?

The idea here would be to ensure that no payment exists before the very end of the cart-completion flow, so I don't think this is a realistic scenario. Let me know if I understood this right?